compiler/wasm/src/noir/noir-wasm-compiler.ts

```diff
@@ -0,0 +1,171 @@
+import { isAbsolute } from 'path';
+
+import { DependencyManager } from './dependencies/dependency-manager';
+import { GithubDependencyResolver as GithubCodeArchiveDependencyResolver } from './dependencies/github-dependency-resolver';
+import { LocalDependencyResolver } from './dependencies/local-dependency-resolver';
+import { FileManager } from './file-manager/file-manager';
+import { Package } from './package';
+import { LogFn } from '../utils';
+import { CompilationResult } from '../types/noir_artifact';
+
+/** Compilation options */
+export type NoirWasmCompileOptions = {
+  /** Logging function */
+  log: LogFn;
+  /** Log debugging information through this function */
+  debugLog: LogFn;
+};
+
+/**
+ * Noir Package Compiler
+ */
+export class NoirWasmCompiler {
+  #log: LogFn;
+  #debugLog: LogFn;
+  #package: Package;
+  /* eslint-disable @typescript-eslint/no-explicit-any */
+  #wasmCompiler: any;
+  #sourceMap: any;
+  /* eslint-disable @typescript-eslint/no-explicit-any */
+  #fm: FileManager;
+  #dependencyManager: DependencyManager;
+
+  private constructor(
+    entrypoint: Package,
+    dependencyManager: DependencyManager,
+    fileManager: FileManager,
+    wasmCompiler: unknown,
+    sourceMap: unknown,
+    opts: NoirWasmCompileOptions,
+  ) {
+    this.#log = opts.log;
+    this.#debugLog = opts.debugLog;
+    this.#package = entrypoint;
+    this.#fm = fileManager;
+    this.#wasmCompiler = wasmCompiler;
+    this.#sourceMap = sourceMap;
+    this.#dependencyManager = dependencyManager;
+  }
+
+  /**
+   * Creates a new compiler instance.
+   * @param fileManager - The file manager to use
+   * @param projectPath - The path to the project
+   * @param opts - Compilation options
+   */
+  public static async new(
+    fileManager: FileManager,
+    projectPath: string,
+    /* eslint-disable @typescript-eslint/no-explicit-any */
+    wasmCompiler: any,
+    sourceMap: any,
+    /* eslint-enable @typescript-eslint/no-explicit-any */
+    opts: NoirWasmCompileOptions,
+  ) {
+    // Assume the filemanager is initialized at the project root
+    if (!isAbsolute(projectPath)) {
+      throw new Error('projectPath must be an absolute path');
+    }
+
+    const noirPackage = await Package.open(projectPath, fileManager);
+
+    const dependencyManager = new DependencyManager(
+      [
+        new LocalDependencyResolver(fileManager),
+        new GithubCodeArchiveDependencyResolver(fileManager),
+        // TODO support actual Git repositories
+      ],
+      noirPackage,
+    );
+
+    return new NoirWasmCompiler(noirPackage, dependencyManager, fileManager, wasmCompiler, sourceMap, opts);
+  }
+
+  /**
+   * Compile EntryPoint
+   */
+  /**
+   * Compile EntryPoint
+   */
+  public async compile(): Promise<CompilationResult[]> {
```

Comment:
```suggestion
  public async compile(): Promise<CompilationResult> {
```

The wasm compiler doesn't return more than one `CompilationResult`

Reply:
Fixed